kie-server-parent/kie-server-services/kie-server-services-prometheus/src/main/java/org/kie/server/services/prometheus/PrometheusKieServerExtension.java

```diff
@@ -93,12 +100,28 @@ public void init(KieServerImpl kieServer, KieServerRegistry registry) {
 
             final KModuleDeploymentService deploymentService = jBPMExtension.getAppComponents(KModuleDeploymentService.class);
             if (deploymentService != null) {
-                deploymentService.addListener(new PrometheusDeploymentEventListener());
+                PrometheusDeploymentEventListener defaultListener = new PrometheusDeploymentEventListener();
```

Comment:
I would suggest maybe just simplifying this bit of code that is repeated in a few places, something like:
```suggestion
                listeners.forEach(l -> deploymentService.addListener(l));
                eploymentService.addListener(new PrometheusDeploymentEventListener());
```

Reply:
Aha great catch @cristianonicolai

Reply:
@cristianonicolai @lucamolteni yes, I'll see how to do that check. I think I have to make these default listeners singletons obtained from PrometheusKieServerExtension. That way we can check if the object is already registered as a listener. I think these listeners work on object identity...